OneCommunityGlobal/HighestGoodNetworkApp
# Kurtis new timer team fixes

<h1 align="center"><img src="https://github.com/0xAbdulKhalid/0xAbdulKhalid/raw/main/assets/mdImages/about_me.gif" width ="25"> Timer Team's awesome timer fix </h1>

<h2>Description:</h2>
<em><strong>Implementation of new timer. </strong></em>Fix Uncaught (in promise) TypeError: Cannot read properties of undefined (reading 'toLowerCase') when submitting tangible time and optimize other .toLowerCase() instances
## Related PRS (if any):
This frontend PR is related to the <a href="https://github.com/OneCommunityGlobal/HGNRest/pull/435
">#435 backend PR</a>.


…

## Main changes explained:
- Implementation of new timer
- fix Uncaught (in promise) TypeError: Cannot read properties of undefined (reading 'toLowerCase') when submitting tangible time
- fix timer auto-restarting after submitting tangible time entry
- fix memory leak
…

## How to test:
1. check into current branch
2. do `npm install` and `...` to run this PR locally
3. log as any role of user
4. start time and stop it to submit a tangible time entry
5. verify functionality and that no errors are occurring besides anything tinyMCE related

## Screenshots or videos of changes:
error in console that I fixed:
![image](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/19482363/e79c9e71-1548-4247-9a48-452ca39fc72d)

timer autorestart fixed:

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/19482363/a02511bd-6322-4cd6-8b74-c93660c0edb7

memory leak fixed:
![image](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/19482363/901eb237-b263-45c3-9d51-ce3bc6299313)

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/19482363/024aa728-e1b8-4b48-a7ce-3383d403152e




## Note:
<em><strong>Issues that are still present:</strong></em>
